### DISCUSSION TITLE
An unexpected situation occurred while using AddArc for testing

### DISCUSSION THREAD

**Discussion by inernoro** [SixLabors/ImageSharp.Drawing#298] [Q&A] [answered]
> I generated some unexpected results when using the following code
Sorry, his center point was accidentally relocated

   ``` [Theory]
    [InlineData(180, 360)]
    [InlineData(180, 359)]
    [InlineData(40, 359)]
    [InlineData(40, 360)]
    [InlineData(10, 359)]
    [InlineData(10, 360)]
    public Image<Rgba32> AddArcTest(int start, int sweep)
    {
        var r = 50;
        var image = new Image<Rgba32>(300, 300, Color.White);
        image.Mutate(x =>
        {
            var centerPoint = new Point(image.Width / 2, image.Height / 2);
            var yuan = new EllipsePolygon(centerPoint.X, centerPoint.Y, r, r);
            x.Draw(Color.Black, 1, yuan);

            var pathBuilder = new PathBuilder();
            var pointF = image.GetCenterPoint();
            pathBuilder.AddArc(pointF, r, r, 0, start, sweep);
            x.Draw(Color.Aqua, 2, pathBuilder.Build());
        });
        image.Save(AsVol($"{start}-{sweep}.png"));
        return image;
    }
```

> 40-360.png

![40-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/4722541a-c3bb-4d42-8368-3b30eb2a4d96)

> 40-359.png

![180-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/4b019644-1e91-424f-ba27-c29953c09b92)

> 180-360.png

![180-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/01326e5e-c529-4970-8a76-12f8030f797b)

> 10-359.png

![10-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/72f82f8d-05ab-4814-8127-892103901c13)

> 10-360 .png

![10-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/1b5a00d7-3ca9-4b60-bdd0-04d549cb5bb4)

> 40-359.png

![40-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/49b295fa-57e6-49af-bffe-9286ee135d91)

**Comment by JimBobSquarePants** [comment]
I've been having a dig through this, and the path builder is producing the correct output each time. There's something wrong with the polygon clipper.

Our clipper is a port of Clipper2 optimized for speed and memory usage and I've checked upstream with the original source and the issue is present there, so I've raised an issue.

**Reply by inernoro** [reply]
It seems that this is an exception rather than an issue with my usage. Has an issue been raised somewhere? I'll go take a look.

**Reply by JimBobSquarePants** [reply]
As I said, I’ve raised an issue with Clipper2. I’ll translate this into an issue so we can track the result there.

**Reply by inernoro** [reply]
I see it, but the issue you submitted appears to have been closed.

**Reply by JimBobSquarePants** [reply, marked as answer]
Yeah, because I was wrong...
I have a solution. I'll convert this now pending the PR.
